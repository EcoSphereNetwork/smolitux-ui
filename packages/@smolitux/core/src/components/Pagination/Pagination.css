.smolitux-pagination {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0.25rem;
}

.smolitux-pagination-item {
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 2.5rem;
  height: 2.5rem;
  padding: 0 0.5rem;
  border-radius: 0.375rem;
  font-size: 0.875rem;
  font-weight: 500;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  user-select: none;
}

/* Variants */
.smolitux-pagination--outlined .smolitux-pagination-item {
  border: 1px solid var(--pagination-border-color, #e5e7eb);
  background-color: var(--pagination-bg, white);
  color: var(--pagination-color, #374151);
}

.smolitux-pagination--outlined .smolitux-pagination-item:hover:not(.smolitux-pagination-item--active):not(.smolitux-pagination-item--disabled) {
  background-color: var(--pagination-hover-bg, #f3f4f6);
  border-color: var(--pagination-hover-border-color, #d1d5db);
}

.smolitux-pagination--outlined .smolitux-pagination-item--active {
  background-color: var(--pagination-active-bg, #3b82f6);
  border-color: var(--pagination-active-border-color, #3b82f6);
  color: var(--pagination-active-color, white);
}

.smolitux-pagination--filled .smolitux-pagination-item {
  background-color: var(--pagination-bg, #f3f4f6);
  color: var(--pagination-color, #374151);
  border: none;
}

.smolitux-pagination--filled .smolitux-pagination-item:hover:not(.smolitux-pagination-item--active):not(.smolitux-pagination-item--disabled) {
  background-color: var(--pagination-hover-bg, #e5e7eb);
}

.smolitux-pagination--filled .smolitux-pagination-item--active {
  background-color: var(--pagination-active-bg, #3b82f6);
  color: var(--pagination-active-color, white);
}

.smolitux-pagination--simple .smolitux-pagination-item {
  background-color: transparent;
  color: var(--pagination-color, #374151);
  border: none;
}

.smolitux-pagination--simple .smolitux-pagination-item:hover:not(.smolitux-pagination-item--active):not(.smolitux-pagination-item--disabled) {
  background-color: var(--pagination-hover-bg, #f3f4f6);
}

.smolitux-pagination--simple .smolitux-pagination-item--active {
  color: var(--pagination-active-color-simple, #3b82f6);
  font-weight: 600;
}

/* Sizes */
.smolitux-pagination--sm .smolitux-pagination-item {
  min-width: 2rem;
  height: 2rem;
  font-size: 0.75rem;
}

.smolitux-pagination--md .smolitux-pagination-item {
  min-width: 2.5rem;
  height: 2.5rem;
  font-size: 0.875rem;
}

.smolitux-pagination--lg .smolitux-pagination-item {
  min-width: 3rem;
  height: 3rem;
  font-size: 1rem;
}

/* Disabled state */
.smolitux-pagination-item--disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.smolitux-pagination--disabled .smolitux-pagination-item {
  opacity: 0.5;
  cursor: not-allowed;
}

.smolitux-pagination--disabled .smolitux-pagination-item:hover {
  background-color: inherit;
  border-color: inherit;
}

/* Ellipsis */
.smolitux-pagination-ellipsis {
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 2.5rem;
  height: 2.5rem;
  color: var(--pagination-ellipsis-color, #6b7280);
  user-select: none;
}

.smolitux-pagination--sm .smolitux-pagination-ellipsis {
  min-width: 2rem;
  height: 2rem;
}

.smolitux-pagination--lg .smolitux-pagination-ellipsis {
  min-width: 3rem;
  height: 3rem;
}

/* Page count */
.smolitux-pagination-count {
  margin-left: 0.75rem;
  font-size: 0.875rem;
  color: var(--pagination-count-color, #6b7280);
}

.smolitux-pagination--sm .smolitux-pagination-count {
  font-size: 0.75rem;
}

.smolitux-pagination--lg .smolitux-pagination-count {
  font-size: 1rem;
}

/* Focus state */
.smolitux-pagination-item:focus {
  outline: none;
}

.smolitux-pagination-item:focus-visible {
  box-shadow: 0 0 0 2px var(--pagination-focus-shadow-color, rgba(59, 130, 246, 0.5));
}

/* Dark mode */
.dark .smolitux-pagination--outlined .smolitux-pagination-item {
  border-color: var(--pagination-border-color-dark, #4b5563);
  background-color: var(--pagination-bg-dark, #1f2937);
  color: var(--pagination-color-dark, #e5e7eb);
}

.dark .smolitux-pagination--outlined .smolitux-pagination-item:hover:not(.smolitux-pagination-item--active):not(.smolitux-pagination-item--disabled) {
  background-color: var(--pagination-hover-bg-dark, #374151);
  border-color: var(--pagination-hover-border-color-dark, #6b7280);
}

.dark .smolitux-pagination--outlined .smolitux-pagination-item--active {
  background-color: var(--pagination-active-bg-dark, #2563eb);
  border-color: var(--pagination-active-border-color-dark, #2563eb);
  color: var(--pagination-active-color-dark, white);
}

.dark .smolitux-pagination--filled .smolitux-pagination-item {
  background-color: var(--pagination-bg-dark, #374151);
  color: var(--pagination-color-dark, #e5e7eb);
}

.dark .smolitux-pagination--filled .smolitux-pagination-item:hover:not(.smolitux-pagination-item--active):not(.smolitux-pagination-item--disabled) {
  background-color: var(--pagination-hover-bg-dark, #4b5563);
}

.dark .smolitux-pagination--filled .smolitux-pagination-item--active {
  background-color: var(--pagination-active-bg-dark, #2563eb);
  color: var(--pagination-active-color-dark, white);
}

.dark .smolitux-pagination--simple .smolitux-pagination-item {
  color: var(--pagination-color-dark, #e5e7eb);
}

.dark .smolitux-pagination--simple .smolitux-pagination-item:hover:not(.smolitux-pagination-item--active):not(.smolitux-pagination-item--disabled) {
  background-color: var(--pagination-hover-bg-dark, #374151);
}

.dark .smolitux-pagination--simple .smolitux-pagination-item--active {
  color: var(--pagination-active-color-simple-dark, #60a5fa);
}

.dark .smolitux-pagination-ellipsis {
  color: var(--pagination-ellipsis-color-dark, #9ca3af);
}

.dark .smolitux-pagination-count {
  color: var(--pagination-count-color-dark, #9ca3af);
}

.dark .smolitux-pagination-item:focus-visible {
  box-shadow: 0 0 0 2px var(--pagination-focus-shadow-color-dark, rgba(96, 165, 250, 0.5));
}

/* Responsive adjustments */
@media (max-width: 640px) {
  .smolitux-pagination {
    gap: 0.125rem;
  }
  
  .smolitux-pagination-item {
    min-width: 2rem;
    height: 2rem;
    padding: 0 0.25rem;
    font-size: 0.75rem;
  }
  
  .smolitux-pagination--lg .smolitux-pagination-item {
    min-width: 2.5rem;
    height: 2.5rem;
    font-size: 0.875rem;
  }
  
  .smolitux-pagination-ellipsis {
    min-width: 2rem;
    height: 2rem;
  }
  
  .smolitux-pagination--lg .smolitux-pagination-ellipsis {
    min-width: 2.5rem;
    height: 2.5rem;
  }
}