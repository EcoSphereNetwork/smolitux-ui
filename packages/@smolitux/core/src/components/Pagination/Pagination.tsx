// packages/@smolitux/core/src/components/Pagination/Pagination.tsx
import React, { forwardRef, useMemo } from 'react';

export interface PaginationProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
  /** Anzahl der Seiten insgesamt */
  pageCount: number;
  /** Aktuelle Seite (1-basiert) */
  currentPage: number;
  /** Callback bei Seitenwechsel */
  onChange: (page: number) => void;
  /** Anzahl der angezeigten Seiten */
  siblingCount?: number;
  /** Zeigt erste/letzte Seite */
  showFirstLast?: boolean;
  /** Zeigt Buttons für vorherige/nächste Seite */
  showPrevNext?: boolean;
  /** Größe der Pagination */
  size?: 'sm' | 'md' | 'lg';
  /** Variante der Pagination */
  variant?: 'outlined' | 'filled' | 'simple';
  /** Deaktiviert die Pagination */
  disabled?: boolean;
  /** Zeigt die Gesamtzahl der Seiten an */
  showPageCount?: boolean;
  /** Benutzerdefinierte Texte */
  labels?: {
    previous?: string;
    next?: string;
    first?: string;
    last?: string;
    page?: string;
  };
  /** Benutzerdefinierte Icons */
  icons?: {
    previous?: React.ReactNode;
    next?: React.ReactNode;
    first?: React.ReactNode;
    last?: React.ReactNode;
  };
}

/**
 * Pagination-Komponente für Seitennavigation
 * 
 * @example
 * ```tsx
 * <Pagination
 *   pageCount={10}
 *   currentPage={1}
 *   onChange={(page) => setCurrentPage(page)}
 * />
 * ```
 */
export const Pagination = forwardRef<HTMLDivElement, PaginationProps>(({
  pageCount,
  currentPage,
  onChange,
  siblingCount = 1,
  showFirstLast = true,
  showPrevNext = true,
  size = 'md',
  variant = 'outlined',
  disabled = false,
  showPageCount = false,
  labels = {
    previous: 'Zurück',
    next: 'Weiter',
    first: 'Erste',
    last: 'Letzte',
    page: 'Seite'
  },
  icons,
  className = '',
  ...rest
}, ref) => {
  // Standardicons für Pagination
  const defaultIcons = {
    previous: (
      <svg className="w-5 h-5" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M15 19l-7-7 7-7" />
      </svg>
    ),
    next: (
      <svg className="w-5 h-5" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M9 5l7 7-7 7" />
      </svg>
    ),
    first: (
      <svg className="w-5 h-5" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M11 19l-7-7 7-7m8 14l-7-7 7-7" />
      </svg>
    ),
    last: (
      <svg className="w-5 h-5" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M13 5l7 7-7 7M5 5l7 7-7 7" />
      </svg>
    )
  };
  
  // Kombiniere Standard- und benutzerdefinierte Icons
  const mergedIcons = { ...defaultIcons, ...icons };
  
  // Größen-spezifische Klassen
  const sizeClasses = {
    sm: 'h-8 min-w-8 text-xs',
    md: 'h-10 min-w-10 text-sm',
    lg: 'h-12 min-w-12 text-base'
  };
  
  // Varianten-spezifische Klassen
  const variantClasses = {
    outlined: {
      default: 'border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700',
      active: 'border border-primary-500 dark:border-primary-400 bg-primary-50 dark:bg-primary-900/20 text-primary-600 dark:text-primary-400',
      disabled: 'border border-gray-200 dark:border-gray-700 bg-gray-100 dark:bg-gray-800 text-gray-400 dark:text-gray-600 cursor-not-allowed'
    },
    filled: {
      default: 'border border-transparent bg-gray-200 dark:bg-gray-700 hover:bg-gray-300 dark:hover:bg-gray-600',
      active: 'border border-transparent bg-primary-500 dark:bg-primary-600 text-white',
      disabled: 'border border-transparent bg-gray-100 dark:bg-gray-800 text-gray-400 dark:text-gray-600 cursor-not-allowed'
    },
    simple: {
      default: 'border-0 bg-transparent hover:text-primary-600 dark:hover:text-primary-400',
      active: 'border-0 bg-transparent text-primary-600 dark:text-primary-400 font-medium',
      disabled: 'border-0 bg-transparent text-gray-400 dark:text-gray-600 cursor-not-allowed'
    }
  };
  
  // Berechnung der angezeigten Seiten
  const paginationItems = useMemo(() => {
    const DOTS = 'dots';
    const totalPageNumbers = siblingCount * 2 + 3; // Anzahl der anzuzeigenden Seiten + erste & letzte
    
    // Wenn weniger Seiten als die anzuzeigenden Seiten vorhanden sind
    if (totalPageNumbers >= pageCount) {
      return Array.from({ length: pageCount }, (_, i) => i + 1);
    }
    
    // Berechnung des ersten und letzten Geschwister-Index
    const leftSiblingIndex = Math.max(currentPage - siblingCount, 1);
    const rightSiblingIndex = Math.min(currentPage + siblingCount, pageCount);
    
    // Ob Ellipsen angezeigt werden sollen
    const shouldShowLeftDots = leftSiblingIndex > 2;
    const shouldShowRightDots = rightSiblingIndex < pageCount - 1;
    
    // Sonderfälle für erste und letzte Seite
    if (!shouldShowLeftDots && shouldShowRightDots) {
      const leftRange = Array.from({ length: 3 + siblingCount * 2 }, (_, i) => i + 1);
      return [...leftRange, DOTS, pageCount];
    }
    
    if (shouldShowLeftDots && !shouldShowRightDots) {
      const rightStart = pageCount - (2 + siblingCount * 2);
      const rightRange = Array.from({ length: 3 + siblingCount * 2 }, (_, i) => rightStart + i + 1);
      return [1, DOTS, ...rightRange];
    }
    
    if (shouldShowLeftDots && shouldShowRightDots) {
      const middleRange = Array.from(
        { length: rightSiblingIndex - leftSiblingIndex + 1 }, 
        (_, i) => leftSiblingIndex + i
      );
      return [1, DOTS, ...middleRange, DOTS, pageCount];
    }
    
    return [];
  }, [currentPage, pageCount, siblingCount]);
  
  // Seitenwechsel-Handler
  const handlePageChange = (page: number) => {
    if (page === currentPage || page < 1 || page > pageCount || disabled) {
      return;
    }
    
    onChange(page);
  };
  
  // Basis-Klassen für Pagination-Items
  const getItemClasses = (isActive: boolean) => [
    'flex items-center justify-center rounded-md',
    sizeClasses[size],
    'transition-colors duration-200 ease-in-out',
    isActive
      ? variantClasses[variant].active
      : disabled
        ? variantClasses[variant].disabled
        : variantClasses[variant].default
  ].join(' ');
  
  return (
    <div
      ref={ref}
      className={`flex items-center space-x-1 ${className}`}
      {...rest}
    >
      {/* Erste Seite */}
      {showFirstLast && (
        <button
          type="button"
          className={getItemClasses(false)}
          onClick={() => handlePageChange(1)}
          disabled={disabled || currentPage === 1}
          aria-label={labels.first}
          title={labels.first}
        >
          {variant === 'simple' ? labels.first : mergedIcons.first}
        </button>
      )}
      
      {/* Vorherige Seite */}
      {showPrevNext && (
        <button
          type="button"
          className={getItemClasses(false)}
          onClick={() => handlePageChange(currentPage - 1)}
          disabled={disabled || currentPage === 1}
          aria-label={labels.previous}
          title={labels.previous}
        >
          {variant === 'simple' ? labels.previous : mergedIcons.previous}
        </button>
      )}
      
      {/* Seiten */}
      {paginationItems.map((item, index) => {
        if (item === 'dots') {
          return (
            <span
              key={`dots-${index}`}
              className={`flex items-center justify-center ${sizeClasses[size]} text-gray-500 dark:text-gray-400`}
            >
              &hellip;
            </span>
          );
        }
        
        const pageNumber = item as number;
        return (
          <button
            key={pageNumber}
            type="button"
            className={getItemClasses(pageNumber === currentPage)}
            onClick={() => handlePageChange(pageNumber)}
            disabled={disabled}
            aria-current={pageNumber === currentPage ? 'page' : undefined}
          >
            {pageNumber}
          </button>
        );
      })}
      
      {/* Nächste Seite */}
      {showPrevNext && (
        <button
          type="button"
          className={getItemClasses(false)}
          onClick={() => handlePageChange(currentPage + 1)}
          disabled={disabled || currentPage === pageCount}
          aria-label={labels.next}
          title={labels.next}
        >
          {variant === 'simple' ? labels.next : mergedIcons.next}
        </button>
      )}
      
      {/* Letzte Seite */}
      {showFirstLast && (
        <button
          type="button"
          className={getItemClasses(false)}
          onClick={() => handlePageChange(pageCount)}
          disabled={disabled || currentPage === pageCount}
          aria-label={labels.last}
          title={labels.last}
        >
          {variant === 'simple' ? labels.last : mergedIcons.last}
        </button>
      )}
      
      {/* Anzeige der Gesamtseitenzahl */}
      {showPageCount && (
        <span className="ml-2 text-sm text-gray-600 dark:text-gray-400">
          {labels.page} {currentPage} / {pageCount}
        </span>
      )}
    </div>
  );
});

Pagination.displayName = 'Pagination';

export default Pagination;
