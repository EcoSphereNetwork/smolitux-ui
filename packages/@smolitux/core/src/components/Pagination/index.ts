// packages/@smolitux/core/src/components/Pagination/index.ts
export { default, type PaginationProps } from './Pagination';
