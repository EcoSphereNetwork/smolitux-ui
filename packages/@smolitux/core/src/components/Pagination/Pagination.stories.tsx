import type { Meta, StoryObj } from '@storybook/react';
import { Pagination } from './Pagination';

const meta: Meta<typeof Pagination> = {
  title: 'Components/Pagination',
  component: Pagination,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'Pagination',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom Pagination',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive Pagination',
    onClick: () => alert('Clicked!'),
  },
};
