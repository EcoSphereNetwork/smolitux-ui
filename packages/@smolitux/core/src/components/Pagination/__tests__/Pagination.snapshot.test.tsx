import React from 'react';
import renderer from 'react-test-renderer';
import { Pagination } from '../Pagination';

describe('Pagination Snapshots', () => {
  it('renders default pagination', () => {
    const tree = renderer.create(<Pagination totalPages={5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with current page and size', () => {
    const tree = renderer.create(<Pagination totalPages={10} currentPage={2} size="sm" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
