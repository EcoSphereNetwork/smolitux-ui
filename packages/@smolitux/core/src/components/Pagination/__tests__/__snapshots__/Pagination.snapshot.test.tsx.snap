// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Snapshots renders default pagination 1`] = `
<div
  aria-label="Pagination"
  className="flex items-center space-x-1 "
  role="navigation"
>
  <button
    aria-label="Erste"
    className="flex items-center justify-center rounded-md h-10 min-w-10 text-sm transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={true}
    onClick={[Function]}
    title="Erste"
    type="button"
  >
    <svg
      className="w-5 h-5"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 19l-7-7 7-7m8 14l-7-7 7-7"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </button>
  <button
    aria-label="Zurück"
    className="flex items-center justify-center rounded-md h-10 min-w-10 text-sm transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={true}
    onClick={[Function]}
    title="Zurück"
    type="button"
  >
    <svg
      className="w-5 h-5"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 19l-7-7 7-7"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </button>
  <button
    aria-current="page"
    className="flex items-center justify-center rounded-md h-10 min-w-10 text-sm transition-colors duration-200 ease-in-out border border-primary-500 dark:border-primary-400 bg-primary-50 dark:bg-primary-900/20 text-primary-600 dark:text-primary-400"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="flex items-center justify-center rounded-md h-10 min-w-10 text-sm transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="flex items-center justify-center rounded-md h-10 min-w-10 text-sm transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="flex items-center justify-center rounded-md h-10 min-w-10 text-sm transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="flex items-center justify-center rounded-md h-10 min-w-10 text-sm transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    aria-label="Weiter"
    className="flex items-center justify-center rounded-md h-10 min-w-10 text-sm transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    title="Weiter"
    type="button"
  >
    <svg
      className="w-5 h-5"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 5l7 7-7 7"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </button>
  <button
    aria-label="Letzte"
    className="flex items-center justify-center rounded-md h-10 min-w-10 text-sm transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    title="Letzte"
    type="button"
  >
    <svg
      className="w-5 h-5"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13 5l7 7-7 7M5 5l7 7-7 7"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`Pagination Snapshots renders with current page and size 1`] = `
<div
  aria-label="Pagination"
  className="flex items-center space-x-1 "
  role="navigation"
>
  <button
    aria-label="Erste"
    className="flex items-center justify-center rounded-md h-8 min-w-8 text-xs transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    title="Erste"
    type="button"
  >
    <svg
      className="w-5 h-5"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 19l-7-7 7-7m8 14l-7-7 7-7"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </button>
  <button
    aria-label="Zurück"
    className="flex items-center justify-center rounded-md h-8 min-w-8 text-xs transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    title="Zurück"
    type="button"
  >
    <svg
      className="w-5 h-5"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 19l-7-7 7-7"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </button>
  <button
    className="flex items-center justify-center rounded-md h-8 min-w-8 text-xs transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    aria-current="page"
    className="flex items-center justify-center rounded-md h-8 min-w-8 text-xs transition-colors duration-200 ease-in-out border border-primary-500 dark:border-primary-400 bg-primary-50 dark:bg-primary-900/20 text-primary-600 dark:text-primary-400"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="flex items-center justify-center rounded-md h-8 min-w-8 text-xs transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="flex items-center justify-center rounded-md h-8 min-w-8 text-xs transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="flex items-center justify-center rounded-md h-8 min-w-8 text-xs transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <span
    className="flex items-center justify-center h-8 min-w-8 text-xs text-gray-500 dark:text-gray-400"
  >
    …
  </span>
  <button
    className="flex items-center justify-center rounded-md h-8 min-w-8 text-xs transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    10
  </button>
  <button
    aria-label="Weiter"
    className="flex items-center justify-center rounded-md h-8 min-w-8 text-xs transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    title="Weiter"
    type="button"
  >
    <svg
      className="w-5 h-5"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 5l7 7-7 7"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </button>
  <button
    aria-label="Letzte"
    className="flex items-center justify-center rounded-md h-8 min-w-8 text-xs transition-colors duration-200 ease-in-out border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 hover:bg-gray-100 dark:hover:bg-gray-700"
    disabled={false}
    onClick={[Function]}
    title="Letzte"
    type="button"
  >
    <svg
      className="w-5 h-5"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13 5l7 7-7 7M5 5l7 7-7 7"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;
